skia/RSkSVGContainerComponent.cpp

```diff
@@ -0,0 +1,145 @@
+
+  static std::unordered_map<std::string, RSkSVGNode* (*)(std::shared_ptr<RSkComponent>)> s_mapStringValues={
+    { "RNSVGView",getRSKSVGNodeFor<RSkComponentRNSVGSvgView>},
+    { "RNSVGGroup",getRSKSVGNodeFor<RSkComponentRNSVGGroup>},
+    { "RNSVGCircle",getRSKSVGNodeFor<RSkComponentRNSVGCircle>},
+    { "RNSVGRect",getRSKSVGNodeFor<RSkComponentRNSVGRect>},
+    { "RNSVGEllipse",getRSKSVGNodeFor<RSkComponentRNSVGEllipse>},
+    { "RNSVGPath",getRSKSVGNodeFor<RSkComponentRNSVGPath>},
+    { "RNSVGLine",getRSKSVGNodeFor<RSkComponentRNSVGLine>},
+    { "RNSVGDefs",getRSKSVGNodeFor<RSkComponentRNSVGDefs>},
+    { "RNSVGUse",getRSKSVGNodeFor<RSkComponentRNSVGUse>}
+  };
+
+  std::string componentName=component->getComponentData().componentName;
+
+  if(s_mapStringValues.find(componentName.c_str()) != s_mapStringValues.end()) {
+    return s_mapStringValues[componentName.c_str()](component);
+  } else {
+    return nullptr;
+  }
+}
+
+void RSkSVGContainerComponent::mountChildComponent(
+    std::shared_ptr<RSkComponent> childComponent,
+    const int index) {
+  
+  RNS_LOG_INFO("Parent Componet :" << this->getComponentData().componentName<<" holding child :" << childComponent->getComponentData().componentName);
+
+  if(!childComponent || !childComponent.get()) {
+    RNS_LOG_ERROR("Invalid child component received");
+    return;
+  }
+
+  auto node=getRSkSVGNodeForComponetWithName(childComponent);
+  if(node) {
+    auto sksvgNode =sk_sp<SkSVGNode>(node);
+    RNS_LOG_INFO("Child Node type  : "<<(int)node->tag());
+    if(!node->nodeName.empty()) {
+    RNS_LOG_INFO("Child Node has ID : "<<node->nodeName);
+    nodeIDMapper_.set(SkString(node->nodeName),sksvgNode);
+    }
+    // If Child Node is of type of container , Merge it's defMap with it's Parent
+    if(DOWN_CAST_TO_CLASS(node,RSkSVGContainerComponent)) {
+      auto mergeFn= [&](SkString keyName,sk_sp<SkSVGNode> *value){
+      RNS_LOG_ERROR("  MERGING child Defs with It's Parent" );
+      nodeIDMapper_.set(keyName,*value);
+      };
+      DOWN_CAST_TO_CLASS(node,RSkSVGContainerComponent)->nodeIDMapper_.foreach(mergeFn);
+    }
+    // Insert child RSkSVGNode in to List
+    size_t indexAt = std::min((size_t)index, childRSkNodeList_.size());
+    RNS_LOG_DEBUG("Insert Child at index : " << indexAt );
+    childRSkNodeList_.insert(childRSkNodeList_.begin() + indexAt, sksvgNode);
+  }
+
+  RNS_LOG_INFO("RSK SVG Def Map Size for component : "<< this->getComponentData().componentName << " : "<<nodeIDMapper_.count());
+  RNS_LOG_INFO("RSK SVG container Size for the Parent : "<<childRSkNodeList_.size());
+}
+
+void RSkSVGContainerComponent::unmountChildComponent(
+    std::shared_ptr<RSkComponent> oldChildComponent,
+    const int index) {
+  RNS_LOG_INFO("RSkSVGContainerComponent recieved unmount for child :" << oldChildComponent->getComponentData().componentName);
+  if(index >= childRSkNodeList_.size()) {
+    RNS_LOG_ERROR("Invalid index passed for remove");
```

Comment:
can we print the index and parent and child component names , it will be helpful during debugging